src/jit_channel/msgs.rs

```diff
@@ -0,0 +1,263 @@
+use std::convert::TryFrom;
+
+use bitcoin::hashes::hmac::{Hmac, HmacEngine};
+use bitcoin::hashes::sha256::Hash as Sha256;
+use bitcoin::hashes::{Hash, HashEngine};
+use serde::{Deserialize, Serialize};
+
+use crate::transport::msgs::{LSPSMessage, RequestId, ResponseError};
+use crate::utils;
+
+pub(crate) const LSPS2_GET_VERSIONS_METHOD_NAME: &str = "lsps2.get_versions";
+pub(crate) const LSPS2_GET_INFO_METHOD_NAME: &str = "lsps2.get_info";
+pub(crate) const LSPS2_BUY_METHOD_NAME: &str = "lsps2.buy";
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize, Default)]
+#[serde(default)]
+pub struct GetVersionsRequest {}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct GetVersionsResponse {
+	pub versions: Vec<u16>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct GetInfoRequest {
+	pub version: u16,
+	pub token: Option<String>,
+}
+
+/// Fees and parameters for a JIT Channel.
+///
+/// The client will pay max(min_fee_msat, proportional*(payment_size_msat/1_000_000)).
+pub struct RawOpeningFeeParams {
+	/// The minimum fee required for the channel open.
+	pub min_fee_msat: u64,
+	/// A fee proportional to the size of the initial payment.
+	pub proportional: u32,
+	/// An ISO8601 formatted date for which these params are valid.
+	pub valid_until: String,
+	/// number of blocks that the LSP promises it will keep the channel alive without closing, after confirmation.
+	pub min_lifetime: u32,
+	/// Maximum number of blocks that the client is allowed to set its to_self_delay parameter.
+	pub max_client_to_self_delay: u32,
+}
+
+impl RawOpeningFeeParams {
+	pub(crate) fn into_opening_fee_params(self, promise_secret: &[u8; 32]) -> OpeningFeeParams {
```

Comment:
Because it needs user to provide the promise_secret, it's not stored in RawOpeningFeeParams. 

The reason we have two of these objects is that we want the LSP to be able to generate their fee params without needing to know anything about the existence of this promise stuff.

To go from the "Raw" to the one that gets sent over the wire we need to use the secret to calculate the promise.